### ISSUE TITLE
result.StandardOutput doesn't receive all stdout

### ISSUE THREAD

**Issue by yrien30** [abbotware/abbotware#2]
I noticed that result.StandardOutput does not receives all the stdout.
According to [MSDN ](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.outputdatareceived?redirectedfrom=MSDN&view=net-5.0): "The application that is processing the asynchronous output should call the WaitForExit method to ensure that the output buffer has been flushed."

Code I used to test this:
```
  IExitInfo result = null;

  string cmd = @"C:\Windows\System32\ftp.exe";
  string args = "-i -s:" + "test.txt" + " " + ipAddress;
  ShellCommandOptions cfg = new ShellCommandOptions(cmd, args)
  {
      CommandTimeout = TimeSpan.FromMinutes(30),
      WorkingDirectory = Directory.GetCurrentDirectory()
  };

  using var cp = new AbbotwareShellCommand(cfg, NullLogger.Instance);

  result = await cp.ExecuteAsync();
```
This code receives all the stdout
```

            string cmd = @"C:\Windows\System32\ftp.exe";
            string args = "-i -s:" + "test.txt" + " " + ipAddress;


            ProcessStartInfo startInfo = new ProcessStartInfo()
            {
                FileName = cmd,
                Arguments = args,
                RedirectStandardOutput = true,
                UseShellExecute = false,
                CreateNoWindow = true,
                WorkingDirectory = Directory.GetCurrentDirectory()
            };

            Process p = new Process();
            p.StartInfo = startInfo;

            p.OutputDataReceived += (sender, e) =>
            {
                if (string.IsNullOrWhiteSpace(e.Data))
                {
                    return;
                }
                richTextBox1.Invoke((MethodInvoker)delegate
                {
                    this.richTextBox1.AppendText(ipAddress + " HMI : " + e.Data + Environment.NewLine);
                });
            };

            p.Start();

            p.BeginOutputReadLine();

            p.WaitForExit(); // This ensures async event handlers are finished processing.
```

**Comment by abbotware** [comment]
@yrien30 - thanks for this -   I made this class to help abstract running commands on windows and Linux - they both seem to behave slightly differently - I have tests that run on both OS's successfully, but let me test out ftp.exe specifically

**Comment by abbotware** [comment]
It seems that I am already putting a small delay to account for the buffer flush - but I can swap that out with WaitForExit(timespan) - it is possible WaitForExit does something additional that a regular delay doesn't account for

**Comment by yrien30** [comment]
> @yrien30 - thanks for this - I made this class to help abstract running commands on windows and Linux - they both seem to behave slightly differently - I have tests that run on both OS's successfully, but let me test out ftp.exe specifically

Would love to use your library if you would add WaitForExit. Because it will not work without it at least not in my case. 

> It seems that I am already putting a small delay to account for the buffer flush - but I can swap that out with WaitForExit(timespan) - it is possible WaitForExit does something additional that a regular delay doesn't account for

Yes, I am pretty sure it does.
According to [MSDN ](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.outputdatareceived?redirectedfrom=MSDN&view=net-5.0): "The application that is processing the asynchronous output should call the WaitForExit method to ensure that the output buffer has been flushed."

**Comment by abbotware** [comment]
Check the latest release - I changed the exit code call WaitForExit - I am still working on creating unit tests for ftp.exe - just curious are you using a script to pass the user/pass? I couldnt seem to find a cmd line option

**Comment by abbotware** [comment]
Also, I should mention that I plan to replace my ILogger with Microsoft.Extensions.Logging.ILogger when i put out a 5.x release of the libraries

**Comment by abbotware** [comment]
@yrien30 - I was able to capture all the output - but I did have to specify **-v** in order to see additional output from the server... not sure why that is the case as it seems counter intuitive. See unit test _WindowsFtpCommand_UbuntuFtpSite_ located here: https://github.com/abbotware/abbotware/blob/master/Test/Abbotware.Interop.IntegrationTests/ShellCommand/WindowsFtpCommandTests.cs

I created a WindowsFtpCommand 'wrapper' to give you insight as to how the class is intended to be used along with an options class. 

https://github.com/abbotware/abbotware/tree/master/Common/Abbotware.ShellCommand/Plugins
Take note of the strongly typed configuration class with attributes that are used to render the command line.

Also, without a script, the ftp.exe program expects input to quit, so there is new overloadable function OnStarted() that shows how you would send input to the standardinput.  (most commands I wrote wrappers for do not expect input so I just added this)

if you use net5, there will be a ctor that takes in the standard microsoft Ilogger
